### PR TITLE
Set up Views, URLs and Templates for basic website structure.

### PR DESCRIPTION
**Pull request by ganttArt**
Boilerplate for Django's MVT (model/view/template) created. These changes add on top of the previously created database, now the project has a front end that you can navigate between a home, about, recipes and grocery planner pages. This includes pages inheriting from the base.html.

### CODE REVIEW

File: groceries_app/views.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
...
+class HomePageView(TemplateView):
+    template_name = 'home.html'
+
+class AboutPageView(TemplateView):
+    template_name = 'about.html'
+
+class RecipesPageView(TemplateView):
+    template_name = 'recipes/recipes.html'
+
+class PlannerPageView(TemplateView):
```
Comment (Azadmel) [inline groceries_app/views.py]: So these are class based views?
What is the naming convention for views? `HomeView` `AboutView` `RecipeView` reads better to me than `HomePageView`

File: project/settings.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'groceries_app.apps.GroceriesAppConfig'
+    'groceries_app',
+    # 'groceries_app.apps.GroceriesAppConfig',
```
Comment (Azadmel) [inline project/settings.py]: Why is this commented out?

File: templates/base.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<!doctype html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+        {% block css %}
+        <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootswatch/4.5.2/sketchy/bootstrap.min.css">
+        {% endblock %}
```
Comment (Azadmel) [inline templates/base.html]: Why does this need to be in block?

File: templates/base.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                        <a class="nav-link" href="recipes">Recipes</a>
+                    </li>
+                    <li class="nav-item active">
+                        <a class="nav-link" href="planner">Grocery Planner</a>
+                    </li>
+                </ul>
+            </div>
+        </nav>
+        {% block content %}
+        {% endblock content %}
```
Comment (Azadmel) [inline templates/base.html:35]: Why empty block here?

File: templates/base.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        
+        {% block javascript %}
+        <!-- Bootstrap JavaScript -->
+        <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
+        <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
+        <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js" integrity="sha384-wfSDF2E50Y2D1uUdj0O3uMBJnjuUD4Ih7YwaYd1iqfktj0Uod8GCExl3Og8ifwB6" crossorigin="anonymous"></script>  
+        {% endblock javascript %}
```
Comment (Azadmel) [inline templates/base.html:42]: Why is this in blocks?

**Review by Azadmel — COMMENTED** [review]
Good start. Left some questions.

**Comment by Azadmel** [comment]
Please add a description.

File: project/settings.py (hunk 2 above)
Comment (ganttArt) [reply project/settings.py]: groceries_app.apps.GroceriesAppConfig was used when I was setting up the database, but this line won't include the groceries app when I need views and urls. By just including 'groceries_app' in my installed apps in settings, the old line became redundant. There is no longer a need to keep the code commented in and I will remove it.

File: groceries_app/views.py (hunk 1 above)
Comment (ganttArt) [reply groceries_app/views.py]: The naming convention for class based views is in line with your suggestion. I will remove Page from these views. Good catch!

File: templates/base.html (hunk 4 above)
Comment (ganttArt) [reply templates/base.html:35]: The empty block here indicates to templates that inherit from this base template, that this is where their block of content will go.

Comment (ganttArt) [reply templates/base.html:35]: https://docs.djangoproject.com/en/3.1/ref/templates/language/#template-inheritance

File: templates/base.html (hunk 3 above)
Comment (ganttArt) [reply templates/base.html]: The formatting of this currently is incorrect for other templates inheriting the css. These stylesheets will be moved outside of the block, but the block will stay so when I add css to other templates inheriting from base.html, that css will be within the html head tag.

**Review by ganttArt — COMMENTED** [review]
Thank you for reviewing the code!

File: templates/base.html (hunk 5 above)
Comment (Azadmel) [reply templates/base.html:42]: Seems like you need update this like the css block update.

**Review by Azadmel — APPROVED** [review]
Looks good.

**Pull request merged**